maven-plugin-tools-annotations/src/main/java/org/apache/maven/tools/plugin/extractor/annotations/JavaAnnotationsMojoDescriptorExtractor.java

```diff
@@ -570,19 +572,18 @@ protected void extendJavaProjectBuilderWithSourcesJar(
             JavaProjectBuilder builder, Artifact artifact, PluginToolsRequest request, String classifier)
             throws ExtractionException {
         try {
-            Artifact sourcesArtifact = repositorySystem.createArtifactWithClassifier(
+            org.eclipse.aether.artifact.Artifact sourcesArtifact = new DefaultArtifact(
                     artifact.getGroupId(),
                     artifact.getArtifactId(),
-                    artifact.getVersion(),
+                    classifier,
                     artifact.getType(),
-                    classifier);
-
-            ArtifactResolutionRequest req = new ArtifactResolutionRequest();
-            req.setArtifact(sourcesArtifact);
-            req.setLocalRepository(request.getLocal());
-            req.setRemoteRepositories(request.getRemoteRepos());
-            ArtifactResolutionResult res = repositorySystem.resolve(req);
-            if (res.hasMissingArtifacts() || res.hasExceptions()) {
+                    artifact.getVersion());
+
+            ArtifactRequest resolveRequest = new ArtifactRequest(sourcesArtifact, request.getRemoteRepos(), null);
```

Comment:
To `ArtifactHandler` required?

Reply:
It is - The context in which this request is made, may be null.

Reply:
My bad, I misread the line. Discard my comment.